starlingcaptureapi/file_util.py

```diff
@@ -56,3 +58,51 @@ def digest_sha256(self, file_path):
                 hasher.update(byte_block)
             return hasher.hexdigest()
         # TODO: handle error (image not found, etc.)
+
+    def cidv1(self, file_path):
+        """Generates the CIDv1 of a file, as determined by ipfs add.
+
+        Args:
+            file_path: the local path to a file
+
+        Returns:
+            CIDv1 in the canonical string format
+
+        Raises:
+            Exception if errors are encountered during processing
+        """
+
+        if not os.path.exists(os.path.expanduser("~/.ipfs")):
+            proc = subprocess.run(
+                ["ipfs", "init"],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                text=True,
+            )
+            if proc.returncode != 0:
+                raise Exception(
+                    f"'ipfs init' failed with code {proc.returncode} and output:\n\n{proc.stdout}"
+                )
```

Comment:
Maybe for a future cleanup: It feels like we could extract the subprocess-related code from here and https://github.com/starlinglab/starling-integrity-api/blob/main/starlingcaptureapi/claim_tool.py#L42-L47 into a helper that takes a `args` list, runs the subprocesses, catches a non-zero return code and excepts if needed, else returns the stdout